### PR TITLE
FEATURE: add site setting for setting locale from header

### PR DESCRIPTION
**Pull request by scossar**
See https://meta.discourse.org/t/different-behaviour-of-text-direction-for-not-logged-in-visitors/40747

Adds a site setting for setting the locale from accept language headers. The new setting defaults to false and is dependent on `allow_user_locale` being set to true.

screenshot 2016-03-14 16 01 05

### CODE REVIEW

**Comment by coding-horror** [comment]
Excellent! Can you review this tomorrow @eviltrout ?

File: lib/validators/allow_user_locale_enabled_validator.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+  def valid_value?(val)
+    # only validate when enabling setting locale from headers
+    return true if val == "f"
+    # ensure that allow_user_locale is enabled
+    SiteSetting.allow_user_locale
+  end
+
+  def error_message
+    "You must first enable 'allow user locale' before enabling this setting."
```
Comment (ZogStriP) [inline lib/validators/allow_user_locale_enabled_validator.rb]: This needs to be translatable ;)

**Comment by scossar** [comment]
The error message is now translatable.

**Comment by SamSaffron** [comment]
thanks

**Pull request merged**